src/Common/src/System/Numerics/Hashing/Hash.cs

```diff
@@ -0,0 +1,96 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace System.Numerics.Hashing
+{
+    internal static class Hash
+    {
+        public static int Combine(int h1, int h2)
+        {
+            return ((h1 << 5) + h1 + (h1 >> 27)) ^ h2;
+        }
```

Comment:
This code looks a lot like   ``` (ROL(h1, 5) + h1) ^ h2 ``` , but not quite that. It could be beneficial to match the ROL pattern since JIT may generate better code for it. - something like:
``` (((h1 << 5) | ((uint)h1 >> 27)) + h1) ^ h2 ``` 

(see: https://github.com/dotnet/coreclr/pull/1830).